### PR TITLE
feat(FN-3389): add docstrings to event handlers and entities

### PR DESCRIPTION
**Pull request by JoshBinns2000**
## Note
This PR is primarily for documentation with minimal modifications to actual implementation. Any comments around implementation are therefore outside the scope of this PR.

## Introduction :pencil2:
We want to add docstrings to event handlers and typeORM entities.

## Resolution :heavy_check_mark:
- Adds docstrings to the event handlers
- Adds docstrings to the entities

### CODE REVIEW

File: libs/common/src/sql-db-entities/utilisation-report/utilisation-report.entity.ts [hunk 1]
```diff
@@ -82,10 +91,18 @@ export class UtilisationReportEntity extends AuditableBaseEntity {
...
+   * Updates the report with fee records
+   * @param param - The parameters to update the report with
+   */
   public updateWithFeeRecords({ feeRecords }: UpdateWithFeeRecordsParams): void {
     this.feeRecords = feeRecords;
   }
 
+  /**
+   * Updates the report with a supplied status
+   * @param param - The parameters to update the report with
```
Comment (abhi-markan) [inline libs/common/src/sql-db-entities/utilisation-report/utilisation-report.entity.ts:104]: Can the parameters names please be reflected, I would encourage adding `@returns` as well if below snippet can be used as a reference please that will be great. Specifying types in TSDocs is optional.

```typescript
/**
 * Updates the report with a supplied status.
 *
 * @param {UpdateWithStatusParams} params - An object containing the status and request source.
 * @param {string} params.status - The new status to update.
 * @param {string} params.requestSource - The source of the update request.
 * @returns {void} - This method does not return a value.
 */
```

File: libs/common/src/sql-db-entities/partial-entities/month-and-year.partial-entity.ts [hunk 2]
```diff
@@ -1,9 +1,15 @@
 import { Column } from 'typeorm';
 
 export class MonthAndYearPartialEntity {
+  /**
```
Comment (abhi-markan) [inline libs/common/src/sql-db-entities/partial-entities/month-and-year.partial-entity.ts:4]: Minor touches to the type.

```typescript
/**
   * The year
   * @required true
   * @example 2024
   */
```

Is the month stored in JS month number (0 - Jan, 11 - Dec) or standard (1-12), an explanation on that will be great. Thanks

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Nice work, minor comments to address please.

File: libs/common/src/sql-db-entities/partial-entities/month-and-year.partial-entity.ts (hunk 2 above)
Comment (JoshBinns2000) [reply libs/common/src/sql-db-entities/partial-entities/month-and-year.partial-entity.ts:4]: As far as I can see, `@required` is not a JSDoc tag. I will make the change documenting it as a one-indexed month though

Comment (abhi-markan) [reply libs/common/src/sql-db-entities/partial-entities/month-and-year.partial-entity.ts:5]: Much cleaner, thanks.

File: libs/common/src/sql-db-entities/utilisation-report/utilisation-report.entity.ts (hunk 1 above)
Comment (abhi-markan) [reply libs/common/src/sql-db-entities/utilisation-report/utilisation-report.entity.ts:111]: Thanks for adding properties to the docstring.
TSDocs reference on `@returns` https://tsdoc.org/pages/tags/returns/, happy to approve on this instance as DocStrings needs to be unified under an established tech-debt ticket.

**Review by abhi-markan — APPROVED** [review]

File: libs/common/src/sql-db-entities/utilisation-report/utilisation-report.entity.ts (hunk 1 above)
Comment (JoshBinns2000) [reply libs/common/src/sql-db-entities/utilisation-report/utilisation-report.entity.ts:111]: That reference highlights that `@returns` documents a return _value_, where `void` is designed to document the absence or a return value, hence the distinction in this case

**Pull request merged**